docs/source/tutorial/understanding_evaluation.rst

```diff
@@ -180,6 +180,75 @@ may be additional known triples :math:`(h, r, t')` for :math:`t \neq t'`. If the
 filtered evaluation setting ignores for a given triple :math:`(h, r, t)` the scores of all other *known* true triples
 :math:`(h, r, t')`.
 
+Below, we present the philosophy from [bordes2013]_ and how it is implemented in PyKEEN:
+
+HPO Scenario
+************
+During training/optimization with :func:`pykeen.hpo.hpo_pipeline`, the set of known positive triples comprises the
+training and validation sets. After optimization is finished and the final evaluation is done, the set of known
+positive triples comprises the training, validation, and testing set. PyKEEN explicitly does not use test triples
+for filtering during HPO avoid any test leakage.
+
+Early Stopper Scenario
+**********************
+When early stopping is used during training, it periodically uses the validation set for calculating the loss
+and evaluation metrics. During this evaluation, the set of known positive triples comprises the training and
+validation sets. When final evaluation is done with the testing set, the set of known positive triples comprises the
+training, validation, and testing set. PyKEEN explicitly does not use test triples for filtering when early stopping
+is being used to avoid any test leakage.
+
+Pipeline Scenario
+*****************
+During vanilla training with the :func:`pykeen.pipeline.pipeline` that has no optimization, no early stopping, nor
+any *post-hoc* choices using the validation set, the set of known positive triples comprises the training and
+testing sets. This scenario is very atypical, and regardless, should be augmented with the validation triples
+to make more comparable to other published results that do not consider this scenario.
+
+Custom Training Loops
+*********************
+In case the validation triples should *not* be filtered when evaluating the test dataset, the argument
+``filter_validation_when_testing=False`` can be passed to either the :func:`pykeen.hpo.hpo_pipeline` or
+:func:`pykeen.pipeline.pipeline`.
+
+When writing a custom training loop and using an :class:`pykeen.evaluation.Evaluator` directly, the set of *known*
+positive triples comprises both the training triples and evaluation triples. However, you can provide additional
```

Comment:
This seems to be a bit unclear.

The (filtered) evaluator will always filter the *evaluation* triples (whether they are validation or testing does not matter from the Evaluator's API). Any additional filter triples need to be passed through the `additional_filter_triples`. 

When used from the pipeline / training loop, they take care of passing training (+validation) to it if required. 

If used stand-alone, the user has to take care of this. This appears to be quite clear from the method's description, but we could mention this here again.

Reply:
Addressed in 33bcbd3b